### ISSUE TITLE
Crash when duplicating `VisualScriptFunctionCall` 

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#53120]
### Godot version

3.4 beta 3

### System information

Ubuntu 21.04, Intel HD 3000

### Issue description

```
	var temp_variable2509 = VisualScriptFunctionCall.new()
	temp_variable2509.set_basic_type(-16)
	temp_variable2509.set_call_mode(3)
	temp_variable2509.duplicate(false)
```
crashes Godot with this backtrace(when compiling with ubsan/asan)
```
modules/visual_script/visual_script_func_nodes.cpp:562:50: runtime error: load of value 4294967280, which is not a valid value for type 'Type'
modules/visual_script/visual_script_func_nodes.cpp:612:47: runtime error: load of value 4294967280, which is not a valid value for type 'Type'
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] godots() [0x17bd100] (/home/rafal/Pulpit/godot/platform/x11/crash_handler_x11.cpp:54)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x41040) [0x7f7f6f0b6040] (??:0)
[3] Map<StringName, _VariantCall::FuncData, Comparator<StringName>, DefaultAllocator>::find(StringName const&) const (/home/rafal/Pulpit/godot/./core/map.h:509)
[4] Variant::get_method_default_arguments(Variant::Type, StringName const&) (/home/rafal/Pulpit/godot/core/variant_call.cpp:1441)
[5] VisualScriptFunctionCall::_validate_property(PropertyInfo&) const (/home/rafal/Pulpit/godot/modules/visual_script/visual_script_func_nodes.cpp:612)
[6] ClassDB::get_property_list(StringName, List<PropertyInfo, DefaultAllocator>*, bool, Object const*) (/home/rafal/Pulpit/godot/core/class_db.cpp:947)
[7] VisualScriptFunctionCall::_get_property_listv(List<PropertyInfo, DefaultAllocator>*, bool) const (/home/rafal/Pulpit/godot/modules/visual_script/visual_script_func_nodes.h:37 (discriminator 15))
[8] Object::get_property_list(List<PropertyInfo, DefaultAllocator>*, bool) const (/home/rafal/Pulpit/godot/core/object.cpp:648)
[9] Resource::duplicate(bool) const (/home/rafal/Pulpit/godot/core/resource.cpp:209)
[10] MethodBind1RC<Ref<Resource>, bool>::call(Object*, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/./core/method_bind.gen.inc:1333 (discriminator 9))
[11] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/core/object.cpp:918 (discriminator 1))
[12] Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) (/home/rafal/Pulpit/godot/core/variant_call.cpp:1173 (discriminator 1))
[13] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript_function.cpp:1037)
[14] GDScriptInstance::_ml_call_reversed(GDScript*, StringName const&, Variant const**, int) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript.cpp:1199)
[15] GDScriptInstance::call_multilevel_reversed(StringName const&, Variant const**, int) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript.cpp:1207)
[16] Node::_notification(int) (/home/rafal/Pulpit/godot/scene/main/node.cpp:149)
[17] Node::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/main/node.h:45 (discriminator 14))
[18] CanvasItem::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/2d/canvas_item.h:162 (discriminator 3))
[19] Node2D::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/2d/node_2d.h:37 (discriminator 3))
[20] Object::notification(int, bool) (/home/rafal/Pulpit/godot/core/object.cpp:929)
[21] Node::_propagate_ready() (/home/rafal/Pulpit/godot/scene/main/node.cpp:188)
[22] Node::_propagate_ready() (/home/rafal/Pulpit/godot/scene/main/node.cpp:178 (discriminator 2))
[23] Node::_set_tree(SceneTree*) (/home/rafal/Pulpit/godot/scene/main/node.cpp:2560)
[24] SceneTree::init() (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:465)
[25] OS_X11::run() (/home/rafal/Pulpit/godot/platform/x11/os_x11.cpp:3635)
[26] godots(main+0x326) [0x17b3d8c] (/home/rafal/Pulpit/godot/platform/x11/godot_x11.cpp:57)
[27] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xd5) [0x7f7f6f09d565] (??:0)
[28] godots(_start+0x2e) [0x17b39ae] (??:?)

```

### Steps to reproduce

above

**Comment by naeu** [comment]
The unsigned enum is being set to -16. Would it be enough to check `p_type`?
```cpp
void VisualScriptFunctionCall::set_basic_type(Variant::Type p_type) {
    if (basic_type == p_type || p_type < 0) {
        return;
```

**Comment by qarmin** [comment]
This was found by fuzzer
Looks that this function is missing checking of `p_type` value
https://github.com/godotengine/godot/blob/7774ec1634374a2c6d10eba4d4c885688747deff/modules/visual_script/visual_script_func_nodes.cpp#L285-L289

**Comment by akien-mga** [comment]
Fixed by #53168.
